src/server/analytic_services/prometheus_reporting.js

```diff
@@ -353,6 +385,22 @@ class PrometheusReporting {
         this._metrics.system_info.set({ system_name: info.name, system_address: info.address }, Date.now());
     }
 
+    set_bucket_status(buckets_info) {
+        if (!this.enabled()) return;
+        buckets_info.forEach(bucket_info => {
```

Comment:
Since it is a metric with dynamic keys you will need to reset it.
Imagine that you have bucket1 and do a set here.
Then you delete the bucket1 and the value will still be in the metrics.
This is why you should reset all of the metrics and set them again for dynamic labels.
You just add the following: ```this._metrics. bucket_healthy.reset();``` after the this.enabled check.